DataDog/datadog-operator
# Pull v0.8 to main

* release v0.8.0
* release v0.8.0-rc.4
* Default Agent version to 7.35.2 (#480)
* release v0.8.0-rc.3
* Add NodeLabelsAsTags and NamespaceLabelsAsTags, remove default Pod env vars (#471) (#475)
* fix: k8s version detection for agent service (#472) (#473)
* release v0.8.0-rc.2
* Update default DCA image to 1.19.0 (#470)
* Add port names to agent service (#468)
* [examples] Fix datadog-agent-secret-backend (#467)
* [controllers/datadogagent/utils] Fix process collection nil check (#466)
* [defaulting/images] Update agent to 7.35.0 (#465)
* release v0.8.0-rc.1
* Update Dockerfile (#464)
* [controllers/datadogagent] Mount certificates volume in DCA (#463)
* tooling: Update golang version to 1.17 (#462)
* Do not enforce dd-agent user (UID 101) for the CLC and DCA (#456)
* update rbac to support secret backend multiple providers script (#459)
* [controllers/datadogagent] Mount tmp volume in DCA and CLC (#461)
* feat: support admissionController.AgentCommunicationMode (#460)
* Fix DatadogAgent secret backend usage (#454)
* add minimum version note to new datadogmonitor examples (#458)
* [controllers/datadogagent] Mount logs volume in DCA and CLC (#455)
* [docs/openshift] Fix example config (#453)
* Add SecurityContext to DCA and CLC (#448)
* update Agent to 7.34.0 and DCA to 1.18.0 (#452)
* [controllers/datadogagent/clusteragent] Fix "externalmetrics" cluster role binding (#451)
* [defaults] Fix cluster agent token autogeneration (#447)
* feat: (orch) support ingresses collection (#450)
* Remove the frequent need to do rm -rf bin to “fix stuff” (#444)
* Add role, role binding, cluster role, cluster role binding and service account to the generated Orchestrator ClusterRole (#390)
* fix: update olm bundles (#445)
* fix(ci): Update codecov action (#446)
* Setting up code scanning on datadog-operator (#442)
* fix: DNSSelectorEndpoints definition (#429)
* [controllers/datadogagent/clusteragent] Add perms to create events (#443)
* Fix continuous create and delete of cluster roles and bindings (#441)
* [controllers/datadogagent] Add datadogmetrics RBACs (#440)
* Fix panic when using minimal recommended dda (#439)
* [controllers/datadogagent/clusteragent] Add subjectaccessreviews to RBAC (#437)
* [defaulting/images] Update latest agent and DCA (#433)
* doc: how to configure API/APP key with OLM (#427)
* [datadogmonitor] Add additional monitor types (#428)
* Improve documentation for the operator in OpenShift (#425)
* update datadog api go client (#426)
* [system-probe] fix chown capability (#420)
* move apis/v1alpha1 utils to apis/utils (#424)
* Set compliance custom benchmarks using config map (#409)
* fix: cluster-agent auth token generation (#410)
* Update pr-linter.yml
* The Network Monitoring feature enables the process agent (#422)
* [v2alpha1] add params to globalconfig (#417)
* Add features to datadogagent v2alpha1 (#413)
* Add missing FQDNs to Cilium network policy
* Add a service with local internal traffic policy (#375)
* Move cilium to cilium/v1
* Add support for Cilium network policies
* fix: add vpa permissions to ksm core rbacs (#411)
* Merge pull request #407 from DataDog/vboulineau/v2notserved
* Update monitor.go (#408)
* Merge branch 'main' into vboulineau/v2notserved
* Merge pull request #406 from DataDog/paulcacheux/update-readme
* Adding Base structure for pod/container config (#405)
* Do not serve v2alpha1 as schema not defined yet
* Update helm chart link to actual chart
* Fix helm chart link in README
* Fix "redhat certified" link in README
* Add documentation on how to test operator bundles (#400)
* Remove defaulting behavior of logs_config.open_files_limit (#396)
* Docs: Update the documentation to use latest v0.7.0 options (#397)
* Merge pull request #394 from DataDog/vboulineau/bump-op
* Only use CRDv1 - Bump operator-sdk version - Add automatic push to RH registries
* Revert "Generate bundles for v1 and v1beta1 separately (#393)"
* Mount /etc/passwd only when needed (#389)
* Generate bundles for v1 and v1beta1 separately (#393)
* Update some docstring fields in CRD (#391)
* Fix RBAC resource name too long (#386)
* Update kubernetes dependencies version + golang (#387)
* Add pod anti-affinity for cluster-agent (#381)
* rm image references in example file (#385)
* rm images in alm-examples (#384)
* update agent and cluster agent image tags (#383)
* Fix: remove Ownerref on Cluster level resources (#377)
* patch bundle to rm status defaultOverride in DatadogAgent (#380)
* fix permissions for extendeddaemonsetreplicasets (#379)
* Merge pull request #378 from DataDog/davidor/fix-old-version-refs
* [controllers/datadogagent] Fix reference to DatadogAgent
* [controllers/datadogagent] Fix watch of ClusterRoles and ClusterRoleBindings (#369)
* remove agent image tags in alm example (#376)
* Introducing v2alpha1 (#374)
* Merge pull request #373 from DataDog/clamoriniere/pull-v0.7-main
* Merge branch 'v0.7' of github.com:DataDog/datadog-operator
* [README] Fix github actions badge (#366)
* [controllers/datadogagent] Ensure ClusterRoles and ClusterRoleBindings are deleted (#359)
* [hack/install-kubebuilder] Fix download link (#360)
* [controllers/datadogagent] Add namespace in "part-of" label (#358)

### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.
